### DISCUSSION TITLE
Use Shader to change vertical scale of terrain

### DISCUSSION THREAD

**Discussion by aiqibaobei** [pelicanmapping/osgearth#2453] [Support Q&A] [answered]
I am using osgEarth to implement terrain exaggeration for an elevation layer, which involves exaggerating the terrain in a specific area. I am aware that the addEffect interface of TerrainEngine can be used to dynamically adjust the scale, but I have found that using this approach exaggerates the entire terrain rather than a specific region. I have applied a C++ shader to the GDALElevationLayer, but it seems to have no effect. Is there another method to achieve my requirement?

**Comment by gwaldron** [comment, marked as answer]
Elevation layers do not support layer shaders. Only Image layers do. You would somehow need to restrict the region from another (image layer) shader.

**Comment by aiqibaobei** [comment]
Thank you so much！

**Comment by horrzs** [comment]
> 我正在使用 osgEarth 为高程图层实现地形夸大，这涉及夸大特定区域的地形。我知道 TerrainEngine 的 addEffect 接口可用于动态调整比例，但我发现使用这种方法会夸大整个地形而不是特定区域。我已将 C++ 着色器应用于 GDALElevationLayer，但它似乎没有效果。有没有其他方法可以满足我的要求？

@aiqibaobei 

> I am using osgEarth to implement terrain exaggeration for an elevation layer, which involves exaggerating the terrain in a specific area. I am aware that the addEffect interface of TerrainEngine can be used to dynamically adjust the scale, but I have found that using this approach exaggerates the entire terrain rather than a specific region. I have applied a C++ shader to the GDALElevationLayer, but it seems to have no effect. Is there another method to achieve my requirement?

@aiqibaobei 请问你说的“TerrainEngine 的 addEffect 接口可用于动态调整地形缩放的比例”，如何实现的？可以给我个实现代码的示例吗？感谢！我在使用osgEarth 3.5
